main/src/ca/uwaterloo/flix/language/phase/Kinder.scala

```diff
@@ -0,0 +1,1155 @@
+/*
+ * Copyright 2021 Matthew Lutze
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.CompilationError
+import ca.uwaterloo.flix.language.ast._
+import ca.uwaterloo.flix.language.errors.KindError
+import ca.uwaterloo.flix.util.Validation
+import ca.uwaterloo.flix.util.Validation.{ToFailure, ToSuccess, flatMapN, mapN, traverse}
+
+/**
+  * Attributes kinds to the types in the AST.
+  *
```

Comment:
What about non-enum, non-defs? E.g. type aliases? classes?

Reply:
Can you elaborate? Enums, classes, instances, type aliases are bundled under the first rule group since their rules are essentially equivalent.

Reply:
Ah yes, but the documentation does not say that. Does it?

Reply:
Line 29 mentions it. Were you looking for something more?

Reply:
My bad!